canonical/rebac-admin
# WD-10180 - feature: add panel for creating roles

## Done

- Add panel for creating roles.
- Update API spec to match the backend.
- Set auth headers.

## QA

- Set `VITE_DEMO_AUTH_TOKEN` to a base64 version of an administrator's username, see [the docs](https://github.com/canonical/identity-platform-admin-ui/wiki/OpenFGA-setup#api-authorization).
- Go to the roles page.
- Click "Create role"
- Fill in the name and submit.
- The form should close and the table should update with the new role.

## Details

- https://warthogs.atlassian.net/browse/WD-10180

## Screenshots

<img width="1433" alt="Screenshot 2024-04-09 at 4 03 16 PM" src="https://github.com/canonical/rebac-admin/assets/361637/b726bb58-5c39-41c8-9d0d-944f0cec9004">
